### PR TITLE
Build http_request_nsurl.mm with ARC

### PR DESCRIPTION
**Pull request by jfirebaugh**
We should build http_request_nsurl.mm with ARC, like the rest of our Objective C code.

Flip [this line](https://github.com/mapbox/mapbox-gl-native/blob/65acf35dcfa3784de7e522926469083068599550/gyp/http-nsurl.gypi#L28) and fix the compile errors.

### CODE REVIEW

**Comment by jfirebaugh** [comment]
👀  @1ec5

**Comment by mb12** [comment]
@jfirebaugh If you want to be superrrr cautious, just do a quick sanity test with leaks in xcode. If there is anything obvious, it would show up.

File: platform/darwin/src/http_request_nsurl.mm [hunk 1]
```diff
@@ -147,11 +126,7 @@
 }
 
 void HTTPNSURLRequest::cancel() {
-    if (task) {
-        [task cancel];
-        [task release];
-        task = nullptr;
-    }
+    [task cancel];
```
Comment (1ec5) [inline platform/darwin/src/http_request_nsurl.mm]: It's good practice to still nil out the task after you cancel it here.

**Comment by jfirebaugh** [comment]
Restored nil assignment; no leaks reported in Instruments.

**Pull request closed without merging**